Packages/com.unity.multiplayer.samples.coop/Utilities/EditorChildSceneLoader.cs

```diff
@@ -0,0 +1,100 @@
+using System;
+using System.Collections.Generic;
+using UnityEditor;
+using UnityEditor.SceneManagement;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+/// <summary>
+/// Should be placed inside root scene and setup with child scenes to load. ChildSceneLoader will take care of the rest
+/// </summary>
+public class EditorChildSceneLoader : MonoBehaviour
+{
+    [SerializeField]
+    public List<SceneAsset> ChildScenesToLoadConfig;
+
+#if UNITY_EDITOR
+    void Update()
+    {
+        // keep this so we can enable/disable this script... (used in ChildSceneLoader)
+    }
+
+    public void SaveSceneSetup()
+    {
+        ChildScenesToLoadConfig ??= new();
+        ChildScenesToLoadConfig.Clear();
+        foreach (var sceneSetup in EditorSceneManager.GetSceneManagerSetup())
+        {
+            ChildScenesToLoadConfig.Add(AssetDatabase.LoadAssetAtPath<SceneAsset>(sceneSetup.path));
+        }
+    }
+
+    public void ResetSceneSetupToConfig()
+    {
+        var sceneAssetsToLoad = ChildScenesToLoadConfig;
+
+        List<SceneSetup> sceneSetupToLoad = new();
+        foreach (var sceneAsset in sceneAssetsToLoad)
+        {
+            sceneSetupToLoad.Add(new SceneSetup() { path = AssetDatabase.GetAssetPath(sceneAsset), isActive = false, isLoaded = true });
+        }
+
+        sceneSetupToLoad[0].isActive = true;
+        EditorSceneManager.SaveCurrentModifiedScenesIfUserWantsTo();
+        EditorSceneManager.RestoreSceneManagerSetup(sceneSetupToLoad.ToArray());
+    }
+#endif
+}
+
+#if UNITY_EDITOR
+
+[CustomEditor(typeof(EditorChildSceneLoader))]
+public class GameEventEditor : Editor
```

Comment:
the name for this class looks like it's a temporary name

Reply:
Oh lol forgot to change that :P 
I should follow my own advice and remove my ctrl, c and v keys from my keyboard...

Reply:
fixed